USER_GUIDE.md

```diff
@@ -0,0 +1,150 @@
+# User Guide
+
+- [Initializing a Client](#initializing-a-client)
+- [Initializing a Client with AWS SigV4 Signing](#initializing-a-client-with-aws-sigv4-signing)
+- [Creating an Index](#creating-an-index)
+- [Add a Document to the Index](#add-a-document-to-the-index)
+- [Search for the Document](#search-for-the-document)
+- [Delete the document](#delete-the-document)
+- [Delete the index](#delete-the-index)
+
+
+## Initializing a Client
+```javascript
+'use strict';
+
+var host = 'localhost';
+var protocol = 'https';
+var port = 9200;
+var auth = 'admin:admin'; // For testing only. Don't store credentials in code.
+var ca_certs_path = '/full/path/to/root-ca.pem';
+
+// Optional client certificates if you don't want to use HTTP basic authentication.
+// var client_cert_path = '/full/path/to/client.pem'
+// var client_key_path = '/full/path/to/client-key.pem'
+
+// Create a client with SSL/TLS enabled.
+var { Client } = require('@opensearch-project/opensearch');
+var fs = require('fs');
+var client = new Client({
+  node: protocol + '://' + auth + '@' + host + ':' + port,
+  ssl: {
+    ca: fs.readFileSync(ca_certs_path),
+    // You can turn off certificate verification (rejectUnauthorized: false) if you're using self-signed certificates with a hostname mismatch.
+    // cert: fs.readFileSync(client_cert_path),
+    // key: fs.readFileSync(client_key_path)
+  },
+});
+```
+
+## Initializing a Client with AWS SigV4 Signing
+
+```javascript
+const endpoint = ""; // OpenSearch domain URL e.g. https://search-xxx.region.es.amazonaws.com
+const { Client } = require('@opensearch-project/opensearch');
+const { AwsSigv4Signer } = require('@opensearch-project/opensearch/aws');
+const { defaultProvider } = require("@aws-sdk/credential-provider-node");
+
+async function getClient() {
+  const credentials = await defaultProvider()();
+  var client = new Client({
+    ...AwsV4Signer({
+      credentials: credentials,
+      region: "us-west-2",
+    }),
+    node: endpoint,
+  });
+  return client;
+}
+```
+
+## Creating an Index
```

Comment:
Pick one or the other in the headings? 

a) "Create an Index", "Add a Document", "Delete ..." 
b) "Creating an Index", "Adding a Document", "Deleting ..."